packages/front-end/components/GeneralSettings/ExperimentSettings/StatsEngineSettings.tsx

```diff
@@ -0,0 +1,399 @@
+import { useEffect, useState } from "react";
+import { useFormContext } from "react-hook-form";
+import { PValueCorrection } from "back-end/types/stats";
+import {
+  DEFAULT_REGRESSION_ADJUSTMENT_DAYS,
+  DEFAULT_SEQUENTIAL_TESTING_TUNING_PARAMETER,
+  DEFAULT_STATS_ENGINE,
+} from "shared/constants";
+import { FaExclamationTriangle } from "react-icons/fa";
+import { useUser } from "@/services/UserContext";
+import { hasFileConfig } from "@/services/env";
+import { GBCuped, GBSequential } from "@/components/Icons";
+import ControlledTabs from "@/components/Tabs/ControlledTabs";
+import Field from "@/components/Forms/Field";
+import PremiumTooltip from "@/components/Marketing/PremiumTooltip";
+import StatsEngineSelect from "@/components/Settings/forms/StatsEngineSelect";
+import SelectField from "@/components/Forms/SelectField";
+import Tab from "@/components/Tabs/Tab";
+import Toggle from "@/components/Forms/Toggle";
+
+export default function StatsEngineSettings() {
+  const form = useFormContext();
+
+  const statsEngine = form.watch("statsEngine");
+  const confidenceLevel = form.watch("confidenceLevel");
+  const pValueThreshold = form.watch("pValueThreshold");
+  const regressionAdjustmentDays = form.watch("regressionAdjustmentDays");
+
+  const { hasCommercialFeature } = useUser();
+  const [statsEngineTab, setStatsEngineTab] = useState<string>(
+    statsEngine || DEFAULT_STATS_ENGINE
+  );
+
+  useEffect(() => {
+    if (statsEngineTab === statsEngine) return;
```

Comment:
You probably don't need the conditional if the dependency is excluded. Once removed, I believe `eslint` might be smart enough to know that `setStatsEngineTab` can also be omitted and you might not need the `eslint-disable`. Could be wrong tho!